packages/valory/skills/decision_maker_abci/behaviours/base.py

```diff
@@ -555,3 +556,97 @@ def finish_behaviour(self, payload: BaseTxPayload) -> Generator:
             yield from self.wait_until_round_end()
 
         self.set_done()
+
+
+class BaseSubscriptionBehaviour(DecisionMakerBaseBehaviour, ABC):
+    """Base class for subscription behaviours."""
+
+    @property
+    def subscription_params(self) -> Dict[str, Any]:
+        """Get the subscription params."""
+        return self.params.mech_to_subscription_params
```

Comment:
Maybe add some sanity checks for the contents of these parameters when retrieving them (just in case). Their keys are being accessed unsafely below.

Reply:
This is one of them as well, its better to have them as you suggest as a dataclass or something similar.

Reply:
#208